econf/econf.py

```diff
@@ -258,77 +268,77 @@ def read_dirs_history(
             err, f"read_dirs_history failed with error: {err_string(err)}"
         )
     arr = cast(key_files, POINTER(c_void_p * c_size.value))
-    result = [c_void_p(i) for i in arr.contents]
+    result = [Keyfile(c_void_p(i)) for i in arr.contents]
     return result
 
 
-def new_key_file(delim: str | bytes, comment: str | bytes) -> c_void_p:
+def new_key_file(delim: str | bytes, comment: str | bytes) -> Keyfile:
     """
     Create a new empty keyfile
 
     :param delim: delimiter of a key/value e.g. '='
     :param comment: string that defines the start of a comment e.g. '#'
-    :return: pointer to created keyfile
+    :return: created Keyfile object
     """
-    result = c_void_p(None)
+    result = Keyfile(c_void_p)
     delim = _encode_str(delim)
     comment = _encode_str(comment)
-    err = libeconf.econf_newKeyFile(result, delim, comment)
+    err = LIBECONF.econf_newKeyFile(byref(result.ptr), delim, comment)
```

Comment:
Is the missing byref a bug in the old code?

Reply:
Yes, from my understanding we need a reference every time we write something into the object. I overlooked this before because I just tested if it threw an error